kernel/vmlambda.ml

```diff
@@ -29,23 +32,52 @@ let is_value lc =
 let get_value lc =
   match lc with
   | Luint i -> val_of_uint i
-  | Lval v -> v
+  | Lval (_, v) -> v
   | Lint i -> val_of_int i
-  | _ -> raise Not_found
+  | _ -> assert false
+
+let hash_block tag args =
+  let open Hashset.Combine in
+  let fold accu v =
+    let h = match v with
+    | Luint i -> Uint63.hash i
+    | Lint i -> Hashtbl.hash (i : int)
```

Comment:
Are we sure the `: int` is enough to get a monomorphic hash? or is the performance good enough anyway?

Reply:
Performance is not the reason I've added the annotation, it is to be sure that the type is known statically.